script.js

```diff
@@ -1,4 +1,150 @@
+
+  // Add 1 to get valid dice rolls of 1 through 6 inclusive.
+  var diceNumber = randomInteger + 1;
+
+  return diceNumber;
+};
+
+/**
+ * Get dice rolls for curr player and populate curr player's dice array
+ * Return the new dice rolls
+ */
+
+var getDiceRolls = function () {
+  // Create an array newDiceRolls with 2 independent dice roll values
+  var newDiceRolls = [getDiceRoll(), getDiceRoll()];
+
+  // Assign newDiceRolls to the current player's dice array
+  if (currPlayer == 1) {
+    player1Dice = newDiceRolls;
+  }
+  // If currPlayer is not 1, assume currPlayer is 2
+  else {
+    player2Dice = newDiceRolls;
+  }
+
+  // Return new dice rolls to parent function
+  return newDiceRolls;
+};
+
+var getPlayerNumber = function (firstNumeralIndex) {
+  // Get the current player's dice array. i.e. var diceArray
+  // if (currPlayer == 1) { diceArray = player1Dice; } else { diceArray = player2Dice; }
+  var diceArray = currPlayer == 1 ? player1Dice : player2Dice;
+  var playerNum;
+  // If the chosen first numeral index is 1, create player number starting with 1st dice
+  if (firstNumeralIndex == 1) {
+    playerNum = Number(String(diceArray[0]) + String(diceArray[1]));
+  }
+  // Otherwise, create player number starting with 2nd dice
+  else {
+    playerNum = Number(String(diceArray[1]) + String(diceArray[0]));
+  }
+
+  // Store player num in the relevant global player num variable
+  if (currPlayer == 1) {
+    player1Num = playerNum;
+  } else {
+    player2Num = playerNum;
+  }
+
+  // Return generated player num to parent function
+  return playerNum;
+};
+
+/**
+ * Compute the winner between Player 1 and Player 2.
+ * Return either 1 or 2 to represent the winning player.
+ * In the event of a tie, Player 2 wins.
+ */
+var determineWinner = function () {
+  if (player1Num > player2Num) {
+    return 1;
+  }
+  return 2;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // Roll 2 dice and show the player the values
+  if (gameMode == GAME_MODE_DICE_ROLL) {
+    // Get dice rolls for curr player and populate the curr player's dice array
+    var newDiceRolls = getDiceRolls();
+    // Switch mode to choose dice order
+    gameMode = GAME_MODE_CHOOSE_DICE_ORDER;
+    // Return the dice roll values to that player
+    return `Welcome Player ${currPlayer}. <br>
+      You rolled ${newDiceRolls[0]} and ${newDiceRolls[1]}. <br><br>
+      Please input 1 or 2 for your preferred order.<br>Option 1: ${newDiceRolls[0]}${newDiceRolls[1]}<br>Option 2: ${newDiceRolls[1]}${newDiceRolls[0]}`;
+  }
+
+  // Create a number based on the player's chosen dice order, and show it to the player
+  if (gameMode == GAME_MODE_CHOOSE_DICE_ORDER) {
+    // Validate the input. If first numeral index is neither 1 nor 2, tell the user.
+    var firstNumeralIndex = Number(input);
+    if (firstNumeralIndex != 1 && firstNumeralIndex != 2) {
+      return "Please input 1 or 2 for your preferred order.";
+    }
```

Comment:
a point of improvement here:
you might wanna consider showing the dice rolls again in this message if they don't enter a valid 1 or 2. Because let's say I did not enter a 1 or 2 and end up getting this message, I might have forgotten already what my dice rolls are and need a reminder. Just another possible enhancement of user experience

Reply:
also another point:
if the dice rolls are the same, does it make sense from a user experience pov to have the user choose the order? think about how we can skip this step and automatically compute the dice roll if they are the same